app/src/main/java/com/leoart/koreanphrasebook/data/repository/DictionaryRepository.kt

```diff
@@ -131,13 +132,22 @@ class DictionaryRepository(val context: Context) : RefreshableRepository {
                         ?: "", it["favourite"] ?: "false"))
             }
         }
+
         isEmpty().observeOn(Schedulers.io())
-                .subscribe({
+                .flatMap {
+                    val syncResult: Single<Boolean>
                     if (it.isSyncNeeded) {
-                        localDB().subscribe { db ->
+                        syncResult = localDB().flatMap { db ->
                             db.dictionaryDao().insertAll(*dictionary.toTypedArray())
-                            DataInfoRepository.getInstance().updateSyncInfo(SyncModel(EDictionary::class.java.simpleName, false))
-                        }
+                            Observable.just(true)
+                        }.single(false)
+                        return@flatMap syncResult
+                    } else {
+                        return@flatMap Single.just(false)
+                    }
+                }.subscribe({
+                    if (it == true) {
+                        DataInfoRepository.getInstance().updateSyncInfo(SyncModel(EDictionary::class.java.simpleName, false))
```

Comment:
@greenhorn31 should be done inside flatmap. It will prevent work on MainThread and it's a nice practice to have a results in subscribe method, not to have any logic